Push to S3

abradley60 opened the pull request:
- Adding function to upload files in a local folder to a specified s3 bucket and folder
- The substructure of the specified folder will be maintained
- Adding a command line tool that can be called 
- Added the default logging level of 'info' to the cli.py script

caitlinadams reviewed (changes requested):
Thanks for this Alex!

Before merging:
* Can you please update the environment.yml file to include boto3 and dependencies? I needed to do `pip install boto3` to get it working
* Perhaps include an option to allow a user to upload a whole folder (e.g. `my-folder/folder bucket my-s3-folder` would create `my-s3-folder/folder` which contained the files)

Looks great and works well otherwise, so happy to approve :)

abradley60 commented:
@caitlinadams the following changes have been made:

- Added dependancies for environment
- renamed the cli from `push_folder_to_s3` -> `upload_files_in_folder_to_s3` for clarity. The `--upload-folder` flag can be set to indicate the whole folder should be uploaded to the specified s3_folder, rather than the files and subfolders inside.

Let me know if any other changes needed
